aqueduct/gateway/tests/utils/mcp.py

```diff
@@ -0,0 +1,192 @@
+    def run(self):
+        from django.conf import settings
+
+        settings.MCP_CONFIG_FILE_PATH = MCP_CONFIG_PATH
+        print(f"Updating MCP_CONFIG_FILE_PATH: {settings.MCP_CONFIG_FILE_PATH}")
+
+        # Configure MCP security settings for testing
+        settings.MCP_ENABLE_DNS_REBINDING_PROTECTION = True
+        settings.MCP_ALLOWED_HOSTS = ["localhost:8000", "localhost:*", "127.0.0.1:*"]
+        settings.MCP_ALLOWED_ORIGINS = [
+            "http://localhost:3000",
+            "http://localhost:*",
+            "http://127.0.0.1:*",
+        ]
+
+        # Allow test hosts in Django's ALLOWED_HOSTS so requests reach our security decorator
+        # Add common test hosts that we'll use to test the security decorator
+        if "evil.testserver" not in settings.ALLOWED_HOSTS:
+            settings.ALLOWED_HOSTS.append("evil.testserver")
+        if "malicious.com" not in settings.ALLOWED_HOSTS:
+            settings.ALLOWED_HOSTS.append("malicious.com")
+
+        print(
+            f"MCP Security: Protection={settings.MCP_ENABLE_DNS_REBINDING_PROTECTION}, Hosts={settings.MCP_ALLOWED_HOSTS}"
+        )
+
+        super().run()
+
+
+class MCPLiveServerTestCase(ChannelsLiveServerTestCase):
+    """
+    Live server test case for MCP endpoints using Django's LiveServerTestCase.
+    Provides a running server for testing HTTP endpoints with httpx.
+    """
+
+    serve_static = False
+    ProtocolServerProcess = MCPDaphneProcess
+
+    fixtures = ["gateway_data.json"]
+    mcp_server_process = None
+
+    @property
+    def headers(self):
+        # ChannelsLiveServerTestCase really hates if I try to access model classes (through import)
+        # -> stops raising AppRegistryNotReady
+        return {"Authorization": "Bearer sk-123abc", "Content-Type": "application/json"}
+
+    @property
+    def mcp_url(self):
+        return f"{self.live_server_url}/mcp-servers/test-server/mcp"
+
+    @asynccontextmanager
+    async def client_session(self):
+        async with streamablehttp_client(self.mcp_url, headers=self.headers) as (
+            read_stream,
+            write_stream,
+            _,
+        ):
+            async with ClientSession(read_stream, write_stream) as session:
+                yield session
+
+    async def assertRequestLogged(self, n: int = 1):
+        # Check that (only) initialize request was logged
+        from management.models import Request
+
+        mcp_requests = await sync_to_async(list)(Request.objects.all())
+        self.assertEqual(len(mcp_requests), n, f"There should be exactly {n} logged MCP request.")
+
+    @classmethod
+    def _write_mcp_config(cls):
+        import json
+
+        os.makedirs(os.path.dirname(MCP_CONFIG_PATH), exist_ok=True)
+        with open(MCP_CONFIG_PATH, "w") as f:
+            json.dump(MCP_TEST_CONFIG, f)
+
+    @classmethod
+    def _update_mcp_config_port(cls, port: int):
+        import json
+
+        config = MCP_TEST_CONFIG.copy()
```

Comment:
Is `copy` enough? Don't we need a `deepcopy` here? (If we don't need a deepcopy, then I guess we also don't need a copy...)